frontend/components/src/content.ts

```diff
@@ -0,0 +1,14 @@
+/* tslint:disable */
+/* eslint-disable */
+// Generated using typescript-generator version 2.26.723 on 2020-11-06 17:30:31.
+
+export interface Content {
+    type: "text";
+}
+
+export interface Text extends Content {
+    type: "text";
+    text: string;
+}
+
+export type ContentUnion = Text;
```

Comment:
With this, we can build a `switch { case }` function in the components lib that takes a `Content` payload and depending on the type field returns a react component.